testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/common/Hydra.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.launcher.env.common;
+
+import com.google.inject.Inject;
+import io.trino.tests.product.launcher.env.DockerContainer;
+import io.trino.tests.product.launcher.env.Environment;
+import io.trino.tests.product.launcher.testcontainers.PortBinder;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import static java.util.Objects.requireNonNull;
+
+public class Hydra
```

Comment:
As i commented elsewhere "Hydra" is meaningless to me, despite me seeing this word around couple times already.

is it possible to give this thing a semantic name? like IdentityProviderSrever or OauthServer or what is it?

Reply:
Hm what do you say about HydraIdentityProvider? For sure, just Hydra gives very little to someone that has not used it, but then hiding the implementation here is a bit too much I think, as tests expect a very specific UI/API, that is Hydra specific.

Reply:
HydraIdentityProvider is good from my perspective.